src/services/refactors/annotateWithTypeFromJSDoc.ts

```diff
@@ -78,7 +78,7 @@ namespace ts.refactor.annotateWithTypeFromJSDoc {
             return Debug.fail(`!decl || !jsdocType || decl.type: !${decl} || !${jsdocType} || ${decl.type}`);
         }
 
-        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const changeTracker = textChanges.ChangeTracker.fromContext(toTextChangesContext(context));

```

Comment:
It looks like every call to `fromContext` calls `toTextChangesContext` first -- would be easier to just do that in `fromContext`, and probably inline `toTextChangesContext`.

Reply:
Great idea!